docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+======================
+Full Stack Application
+======================
+
+Overview
+========
+In this tutorial, we’ll walk you through using Smithy in a full-stack application. This includes defining a model for
+a simple coffee-shop service, generating TypeScript code for both the client and server, and implementing both a
+front-end and back-end for the service. We won't assume that you're an expert in Smithy, but it may helpful to work
+through the :doc:`../quickstart` before beginning this tutorial.
+
+Let's get started.
+
+Setting Up the Project
+======================
+This application will consist of a 4 major components:
+
+1. The model
+2. The server
+3. The client
+4. The web application
+
+--------------
+Pre-requisites
+--------------
+To follow this tutorial, you'll need to install a few tools.
+
+* :doc:`Smithy CLI <../guides/smithy-cli/cli_installation>`
+* `Node.js (>= 16) <https://nodejs.org/en/download>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_
+
+------
+Set up
```

Comment:
Nit: I wonder if each of these should have a separate page? I think that might be slightly easier to follow.

Reply:
IMO, we should try and keep tutorials short and crisp. I think that multiple pages will lead to tutorials that are too long and unfocused. 

What a "single page" constitutes is up for debate, but I think anywhere from 1000 to 3000 words (excluding code) is acceptable. Any longer than that, we probably need to refocus or scope down a tutorial.